### PR TITLE
[TIPC] Add scripts for npu and xpu, test=develop

### PR DESCRIPTION
**Pull request by YanhuiDua**
### PR types
New features

### PR changes
Others 

### Description

为昇腾和昆仑设备添加 TIPC 脚本:

#### 准备小数据集
`bash test_tipc/prepare.sh ./test_tipc/configs/bigru_crf/train_infer_python.txt 'lite_train_lite_infer'
`
#### NPU上运行单测模型，注意这里的脚本带 _npu.sh 后缀
```
bash test_tipc/test_train_inference_python_npu.sh \
         ./test_tipc/configs/bigru_crf/train_infer_python.txt 'lite_train_lite_infer'

```
#### XPU上运行单测模型，注意这里的脚本带 _xpu.sh 后缀
```
bash test_tipc/test_train_inference_python_xpu.sh \
         ./test_tipc/configs/bigru_crf/train_infer_python.txt 'lite_train_lite_infer'

```

### CODE REVIEW

File: examples/machine_translation/transformer/deploy/python/inference.py [hunk 1]
```diff
@@ -131,7 +131,9 @@ def create_predictor(cls,
             if args.device == "gpu":
                 config.enable_use_gpu(100, 0)
             elif args.device == "xpu":
-                config.enable_xpu(100)
+                config.enable_xpu()
+            elif args.device == "npu":
+                config.enable_npu()
```
Comment (YanhuiDua) [inline examples/machine_translation/transformer/deploy/python/inference.py:150]: add npu device

File: examples/machine_translation/transformer/predict.py [hunk 2]
```diff
@@ -60,6 +60,13 @@ def parse_args():
         type=str,
         help="The eos token. It should be provided when use custom vocab_file. "
     )
+    parser.add_argument(
+        "--device", 
+        default="gpu", 
+        choices=["gpu", "cpu", "xpu", "npu"], 
+        help="Device selected for inference."
+    )
+
```
Comment (YanhuiDua) [inline examples/machine_translation/transformer/predict.py:81]: add arguement "--device" to enable npu/xpu

File: examples/machine_translation/transformer/predict.py [hunk 3]
```diff
@@ -157,6 +168,7 @@ def do_predict(args):
     args.unk_token = ARGS.unk_token
     args.bos_token = ARGS.bos_token
     args.eos_token = ARGS.eos_token
+    args.device = ARGS.device
```
Comment (YanhuiDua) [inline examples/machine_translation/transformer/predict.py:183]: transform ARGS.device to args.device

File: tests/test_tipc/test_train_inference_python_npu.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+BASEDIR=$(dirname "$0")
+REPO_ROOT_PATH=$(readlinkf ${BASEDIR}/../)
+
+FILENAME=$1
+
+# change gpu to npu in tipc txt configs
+sed -i "s/--device:gpu/--device:npu/g" $FILENAME
+sed -i "s/state=GPU/state=NPU/g" $FILENAME
+sed -i "s/trainer:pact_train/trainer:norm_train/g" $FILENAME
+sed -i "s/trainer:fpgm_train/trainer:norm_train/g" $FILENAME
```
Comment (YanhuiDua) [inline tests/test_tipc/test_train_inference_python_npu.sh:41]: change "pact_train" or "fpgm_train" to "norm_train"

**Review by ZeyuChen — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**